spec/sierra_api_client_spec.rb

```diff
@@ -236,4 +231,39 @@
       expect(resp.code).to eq(500)
     end
   end
+
+  describe :refresh_oauth do
+    before do
+      stub_request(:get, "#{ENV['SIERRA_API_BASE_URL']}one-reattempt").to_return(
+        { status: 401, body: '{ "foo": "bar1" }' },
+        { status: 200, body: '{ "foo": "bar1" }' },
+      )
+
+      stub_request(:get, "#{ENV['SIERRA_API_BASE_URL']}maximum-attempts-path").to_return(status: 401, body: '{ "foo": "bar1" }')
+
+      stub_request(:post, "#{ENV['SIERRA_OAUTH_URL']}")
+        .to_return(
+          { status: 200, body: '{ "access_token": "fake-access-token" }' },
+          { status: 200, body: '{ "access_token": "second-fake-access-token" }' },
+        )
+    end
+
+    it "should refresh oauth token for 401" do
+      client = SierraApiClient.new
+      first_token = client.instance_variable_get(:@access_token)
+
+      resp = client.get('one-reattempt')
+      second_token = client.instance_variable_get(:@access_token)
+
+      expect(first_token).not_to eq(second_token)
+      expect(client.instance_variable_get(:@retries)).to eq(0)
+      expect(resp).to be_a(SierraApiResponse)
+    end
+
+    it "should throw error once maximum retries attempted" do
+      client = SierraApiClient.new
+      expect { client.get('maximum-attempts-path') }.to raise_error(SierraApiClientTokenError)
+      expect(client.instance_variable_get(:@retries)).to eq(3)
```

Comment:
Related to comment on line 108, note that this test confirms the client is left in a state of having maxed out retries. If the very next call is successful, it's reset. But it should also be reset when the max is hit so that a subsequent call (potentially made some time later) gets the full number of retries.